net/tcp/tcp_recvfrom.c

```diff
@@ -706,7 +706,8 @@ ssize_t psock_tcp_recvfrom(FAR struct socket *psock, FAR struct msghdr *msg,
    *    want for more because there may be no timeout).
    */
 
-  if (state.ir_recvlen == 0 && state.ir_buflen > 0)
+  if (((flags & MSG_WAITALL) != 0 || state.ir_recvlen == 0) &&
```

Comment:
need change line 415 too

Reply:
if some bytes receive, we shouldn't return error at line 730 and 740

Reply:
> if some bytes receive, we shouldn't return error at line 730 and 740

I guess this is true for line 736 too?

Reply:
@xiaoxiang781216 Can you please have a look?